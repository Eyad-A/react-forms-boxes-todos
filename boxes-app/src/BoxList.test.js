import React from "react";
import { render, fireEvent, waitFor } from "@testing-library/react";
import BoxList from "./BoxList";

function addBox(boxList, height = "150", width = "150", color = "purple") {
    const heightInput = boxList.getByLabelText("Height");
    const widthInput = boxList.getByLabelText("Width");
    const backgroundInput = boxList.getByLabelText("Background Color");
    fireEvent.change(backgroundInput, { target: { value: color } });
    fireEvent.change(widthInput, { target: { value: width } });
    fireEvent.change(heightInput, { target: { value: height } });
    const button = boxList.getByText("Add a new box");
    fireEvent.click(button);
}

it("should render without crashing", () => {
    render(<BoxList />);
});

it("should match the snapshot", () => {
    const { asFragment } = render(<BoxList />);
    expect(asFragment()).toMatchSnapshot();
});

it("should add a new box", () => {
    const boxList = render(<BoxList />);

    // expect to see no boxes
    expect(boxList.queryByText("X")).not.toBeInTheDocument();
    addBox(boxList);

    // expect to see one box 
    const removeButton = boxList.getByText("X");
    expect(removeButton).toBeInTheDocument();
    expect(removeButton.previousSibling).toHaveStyle(`
        width: 150px;
        height: 150px;
        background-color: purple;
    `);
});

