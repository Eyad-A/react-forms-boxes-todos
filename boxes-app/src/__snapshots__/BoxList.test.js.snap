// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <div>
        <label
          for="backgroundColor"
        >
          Background Color
        </label>
        <input
          id="backgroundColor"
          name="backgroundColor"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="height"
        >
          Height
        </label>
        <input
          id="height"
          name="height"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="width"
        >
          Width
        </label>
        <input
          id="width"
          name="width"
          type="text"
          value=""
        />
      </div>
      <button>
        Add a new box
      </button>
    </form>
  </div>
</DocumentFragment>
`;
